module/VuFind/src/VuFind/Security/CspHeaderGenerator.php

```diff
@@ -124,4 +132,41 @@ protected function createHeaderObject()
             ? new ContentSecurityPolicyReportOnly()
             : new ContentSecurityPolicy();
     }
+
+    /**
+     * Create Report-To header based on given configuration
+     *
+     * @return GenericHeader
+     */
+    public function getReportToHeader()
+    {
+        $reportToHeader = new GenericHeader();
+        $reportToHeader->setFieldName('Report-To');
+        $groupsText = [];
+
+        $reportTo = $this->config->ReportTo;
+        foreach ($reportTo['groups'] ?? [] as $groupName) {
+            $configSectionName = 'ReportTo' . $groupName;
+            $groupConfig = $this->config->$configSectionName ?? false;
+            if ($groupConfig) {
+                $group = [
+                    'group' => $groupName,
+                    'max-age' => $groupConfig->max_age ?? 86400, // one day
+                    'endpoints' => [],
+                ];
+                foreach ($groupConfig->endpoints_url ?? [] as $url) {
+                    $group['endpoints'][] = [
+                        'url' => $url,
+                    ];
+                }
+                $groupsText[] = json_encode($group, JSON_UNESCAPED_SLASHES);
+            }
+        }
+
+        if (!$groupsText) {
+            return false;
```

Comment:
Maybe return null here instead of false, and change the type above to ?GenericHeader.

Reply:
(On a related note, I notice that the existing return type on `getHeader` needs a `?` added to the front to account for its nullability.

Reply:
Done.